src/ipaddress.h

```diff
@@ -0,0 +1,57 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef IPADDRESS_H
+#define IPADDRESS_H
+
+#include <QHostAddress>
+
+class IPAddress final {
+ public:
+  static IPAddress create(const QString& ip);
+  static QList<IPAddress> excludes(const QList<IPAddress>& sourceList,
```

Comment:
`exclude` seems a little better to me than `excludes`. I would expect `excludes` to be the opposite of `contains`

Reply:
I did a couple of important changes:
- fix the allowed ipv4 lists - now all works as expected
- I renamed `excludes` to `excludeAddresses`
- I kept `contains()` and `overlaps()` because they are methods of the object, used internally and they follow the QT name convention. Ex: `QString::contains()`